src/eredis_client.erl

```diff
@@ -231,10 +234,17 @@ handle_info({reconnect, _Reason}, State) ->
 handle_info(_Info, State) ->
     {noreply, State}.
 
-terminate(_Reason, #state{socket = undefined}) ->
+terminate(_Reason, #state{socket = undefined, sentinel = undefined}) ->
     ok;
-terminate(_Reason, #state{socket = Socket, transport = Transport}) ->
-    Transport:close(Socket).
+terminate(_Reason, #state{socket = Socket, transport = Transport, sentinel = undefined}) ->
+    Transport:close(Socket);
+terminate(_Reason, #state{socket = undefined, sentinel = SentinelOptions}) ->
+    MasterGroup = proplists:get_value(master_group, SentinelOptions, mymaster),
+    eredis_sentinel:stop(MasterGroup);
+terminate(_Reason, #state{socket = Socket, transport = Transport, sentinel = SentinelOptions}) ->
+    Transport:close(Socket),
+    MasterGroup = proplists:get_value(master_group, SentinelOptions, mymaster),
+    eredis_sentinel:stop(MasterGroup).
```

Comment:
Here we stop the sentinel for this master group, even if it was not started by this eredis client process. It means this affects other eredis client processes.

Maybe it works but it is strange.

How about we skip this and let the eredis_sentinel process handle the 'EXIT' signal when the linked eredis_client exits?